src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/CollectionSortClauseBuilder.java

```diff
@@ -102,6 +102,39 @@ public SortClause buildClauseFromDefinition(ObjectNode sortNode) {
     return new SortClause(sortExpressions);
   }
 
+  /**
+   * Helper method to build a "non-special" sort expression for given definition; validates
+   * expression value and builds the {@link SortExpression} object.
+   *
+   * @param path Path to the field to sort by, already validated
+   * @param innerValue JSON value of the sort expression to use
+   * @return {@link SortExpression} for the regular sort
+   */
+  private SortExpression buildRegularSortExpression(String path, JsonNode innerValue) {
+    if (!innerValue.isInt()) {
+      // Special checking for String and ArrayNode to give less confusing error messages
+      if (innerValue.isArray()) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE_VALUE.toApiException(
```

Comment:
Note: existing ErrorV1, not adding any more -- table-side written using ErrorV2.